frontend/webapp/utils/ldap/forms.py

```diff
@@ -0,0 +1,106 @@
+"""Provides Flask-Security login forms for usage with LDAP auth backend."""
+from flask import current_app
+from flask import request
+from flask_security.confirmable import requires_confirmation
+from flask_security.forms import Form
+from flask_security.forms import get_form_field_label
+from flask_security.forms import NextFormMixin
+from flask_security.utils import config_value
+from flask_security.utils import encrypt_password
+from flask_security.utils import get_message
+from flask_security.utils import verify_and_update_password
+from ldap3.core.exceptions import LDAPExceptionError
+from werkzeug.local import LocalProxy
+from wtforms import BooleanField
+from wtforms import PasswordField
+from wtforms import StringField
+from wtforms import SubmitField
+
+_datastore = LocalProxy(lambda: current_app.extensions["security"].datastore)
+
+
+class LDAPLoginForm(Form, NextFormMixin):
+    """Login form for LDAP users."""
+
+    email = StringField("User ID")
+    password = PasswordField(get_form_field_label("password"))
+    remember = BooleanField(get_form_field_label("remember_me"))
+    submit = SubmitField(get_form_field_label("login"))
+
+    def __init__(self, *args, **kwargs):
+        """Init new LDAP login form."""
+        super(LDAPLoginForm, self).__init__(*args, **kwargs)
+        self._args = args
+        self._kwargs = kwargs
+        if not self.next.data:
+            self.next.data = request.args.get("next", "")
+        self.remember.default = config_value("DEFAULT_REMEMBER_ME")
+
+    def validate(self):
+        """Validate LDAP logins against AD."""
+        if not super(LDAPLoginForm, self).validate():
+            return False
+
+        if self.email.data.strip() == "":
+            self.email.errors.append(get_message("USERID_NOT_PROVIDED")[0])
+            return False
+
+        if self.password.data.strip() == "":
+            self.password.errors.append(get_message("PASSWORD_NOT_PROVIDED")[0])
+            return False
+
+        try:
+            # first we try authenticating against ldap
+            user_dn, ldap_data = _datastore.query_ldap_user(self.email.data)
+
+            if not _datastore.verify_password(user_dn, self.password.data):
+                self.password.errors.append(get_message("INVALID_PASSWORD")[0])
```

Comment:
I think that in this case, you also need to delete the current user from the local datastore. Same if the user id cannot be found in LDAP. ATTENTION: you should never delete the local superuser.